aiohttp/web_fileresponse.py

```diff
@@ -377,11 +394,16 @@ async def _prepare_open_file(
         # extension of the request path. The encoding returned by guess_type
         #  can be ignored since the map was cleared above.
         if hdrs.CONTENT_TYPE not in self._headers:
-            if sys.version_info >= (3, 13):
-                guesser = CONTENT_TYPES.guess_file_type
+            if self._path is not None:
+                if sys.version_info >= (3, 13):
+                    guesser = CONTENT_TYPES.guess_file_type
+                else:
+                    guesser = CONTENT_TYPES.guess_type
+                self.content_type = guesser(self._path)[0] or FALLBACK_CONTENT_TYPE
             else:
-                guesser = CONTENT_TYPES.guess_type
-            self.content_type = guesser(self._path)[0] or FALLBACK_CONTENT_TYPE
+                # content-type cannot be determined if self._io is used
+                # instead of self._path
+                self.content_type = FALLBACK_CONTENT_TYPE
```

Comment:
Unless I'm missing something, it seems like FileResponse is missing Content-Disposition by default. If we were to add this in and recommend the user to specify a filename for the download, then we could use that filename here (or just use it if the header has been explicitly passed):
```
>>> mimetypes.guess_type("foo.html")
('text/html', None)
```